components/supervisor/frontend/src/ide/ide-metrics-service-client.ts

```diff
@@ -40,4 +49,32 @@ export class IDEMetricsServiceClient {
             return false
         }
     }
+
+    static async reportError(error: Error, workspaceId: string, instanceId: string, userId: string) : Promise<boolean> {
+        const url = `${MetricsUrl}/reportError`
+        const params: ReportErrorParam = {
+            errorStack: error.stack ?? "",
```

Comment:
I'm not sure it is enough, there is also `message` even if `stack` is missing, also as a complete fallback there is `toString`. I am not sure there are these options are mutually exclusive.

Reply:
just fyi browser error telemetry in VS Code: 
- https://github.com/gitpod-io/openvscode-server/blob/ffd06f4d409bae03df48a3ad2e61f1523605fd20/src/vs/platform/telemetry/browser/errorTelemetry.ts#L11
- https://github.com/gitpod-io/openvscode-server/blob/ffd06f4d409bae03df48a3ad2e61f1523605fd20/src/vs/platform/telemetry/common/errorTelemetry.ts#L47

they seem to track msg and stack separately, they can also track line and column, not sure why we don't get this info, maybe we are instrumenting wrong points

Reply:
interesting that they are buffer, remove duplicates and flush after 5s, not sure whether we need to do it as well now

Reply:
I don't think browser guruantee that `stack` is always present and that `message` is first line. I think we must collect both. They can be repetition, but it won't be good if we don't collect or if message is different.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack